Library/Homebrew/formula.rb

```diff
@@ -3102,6 +3110,56 @@ def license(args = nil)
       end
     end
 
+    # @!attribute [w] allow_network_access
+    # The phases for which network access is allowed. By default, network
+    # access is allowed for all phases. Valid phases are `:build`, `:test`,
+    # and `:postinstall`. When no argument is passed, network access will be
+    # allowed for all phases.
+    # <pre>allow_network_access</pre>
+    # <pre>allow_network_access :build</pre>
+    # <pre>allow_network_access [:build, :test]</pre>
+    def allow_network_access(phases = [])
```

Comment:
A `sig` for these new methods would be nice.

Reply:
I had a think about how to `sig` these; unfortunately, if we add typing on this then we cannot accept plain symbols as arguments. This is because [Sorbet doesn't support literal types](https://github.com/sorbet/sorbet/issues/2504) (related - https://blog.jez.io/tenum-pro-con/).

Instead the argument would be an instance of a `T::Enum` and calling it might look something like `deny_network_access NetworkAccessPhase.Build` or `allow_network_access [NetworkAccessPhase.Test, NetworkAccessPhase.Postinstall]`.

Reply:
@alebcay Could this not be `phases: T.any(T::Array[Symbol], Symbol)`?

> calling it might look something like `deny_network_access NetworkAccessPhase.Build` or `allow_network_access [NetworkAccessPhase.Test, NetworkAccessPhase.Postinstall]`.

Would rather avoid this in formulae.

Reply:
> @alebcay Could this not be `phases: T.any(T::Array[Symbol], Symbol)`?
> 

Strangely this is working here... not sure why something similar didn't work when I was trying to add types in `sandbox.rb`. Will try and re-visit that there too.

Reply:
I've changed the sigs in `formula.rb` and also adjusted some sigs in `sandbox.rb` to remove the need for `T::Enum`.

One slight downside is that Sorbet will not check if a given symbol is a valid argument or not, but that's not an assurance we had without typing, anyways.